doc/developers/auth/jwt.rst

```diff
@@ -0,0 +1,139 @@
+
+Storing public token signing keys
+---------------------------------
+
+IdPs sign tokens with private keys that are periodically updated.
+Corresponding public keys in the form of certificates are needed to validate the signature of a token and are typically published by IdPs at a well-known location in JWKS (`JSON Web Key Set <https://tools.ietf.org/html/rfc7517>`_) format.
+
+.. note::
+
+    Most IdPs support the OpenID Connect Discovery specification which defines how to discover the location to the  certificates.
+    In short, the IdP issuer URL, for example ``https://login.microsoftonline.com/common/v2.0``, is suffixed with ``/.well-known/openid-configuration`` and the ``jwks_uri`` field of the JSON document at that URL is the location of the JWKS     document containing the certificates currently in use. In this example, it would be `<https://login.microsoftonline.com/    common/discovery/v2.0/keys>`_.
+
+Before adding public token signing keys to a running CCF network, the IdP has to be stored as token issuer with a ``set_jwt_issuer`` proposal:
+
+.. code-block:: bash
+
+    $ cat issuer.json
+    {
+      "issuer": "https://login.microsoftonline.com/common/v2.0"
+    }
+    $ python -m ccf.proposal_generator set_jwt_issuer issuer.json
+
+Note that ``issuer.json`` has some additional optional fields for more advanced scenarios.
+See :ref:`developers/auth/jwt:Advanced issuer configuration` for details.
+
+After this proposal is accepted, signing keys for an issuer can be updated with a ``set_jwt_public_signing_keys`` proposal:
+
+.. code-block:: bash
+
+    $ curl "https://login.microjsononline.com/common/discovery/v2.0/keys" -o jwks.json
+    $ ISSUER="https://login.microsoftonline.com/common/v2.0"
+    $ python -m ccf.proposal_generator set_jwt_public_signing_keys $ISSUER jwks.json
+
+If an issuer should not be used anymore, then a ``remove_jwt_issuer`` proposal can be used to remove both the issuer and its signing keys:
+
+.. code-block:: bash
+
+    $ ISSUER="https://login.microsoftonline.com/common/v2.0"
+    $ python -m ccf.proposal_generator remove_jwt_issuer $ISSUER
+
+Validating tokens
+-----------------
+
+Validating a token means checking its format, signature, and IdP- and app-specific claims.
+See `samples/apps/forum/src/authentication.ts <https://github.com/microsoft/CCF/blob/master/samples/apps/forum/src/authentication.ts>`_ for an example on how to do this in TypeScript.
+
+Token signing keys are stored in the ``public:ccf.gov.jwt_public_signing_keys`` kv map where the key is the key ID and the value the DER-encoded X.509 certificate. The key ID matches the ``kid`` field in the token header and can be used to retrieve the matching certificate for validation.
+
+If an application uses multiple token issuers, then the ``public:ccf.gov.jwt_public_signing_key_issuer`` kv map which maps key IDs to issuers can be used to determine the issuer that a key belongs to.
+
+Example using Microsoft Identity Platform
```

Comment:
Could we move this section to its own page? I've found it much easier to maintain/refactor our documentation with shorter doc pages. I believe it's also better for users/developers if they don't have to deal with long documentation pages.

Reply:
Where should it go exactly?

Reply:
In the `auth/` folder, e.g. `auth/jwt_msft_example.rst`?

Reply:
Done